include/api.php

```diff
@@ -5972,6 +5999,42 @@ function api_saved_searches_list($type)
 /// @TODO move to top of file or somewhere better
 api_register_func('api/saved_searches/list', 'api_saved_searches_list', true);
 
+/*
+ * Bind comment numbers(friendica_comments: Int) on each statuses page of *_timeline / favorites / search
+ *
+ * @brief Number of comments
+ *
+ * @param object $data [Status, Status]
+ *
+ * @return void
+ */
+function bindComments(&$data) 
+{
+	if (count($data) == 0) {
+		return;
+	}
+	
+	$ids = [];
+	$comments = [];
+	foreach ($data as $item) {
+		$ids[] = $item['id'];
+	}
+
+	$idStr = DBA::escape(implode(", ", $ids));
+	$sql = "SELECT `parent`, COUNT(*) as comments FROM `item` WHERE `parent` IN ($idStr) AND `deleted` = ? AND `gravity`= ? GROUP BY `parent`";
```

Comment:
This query is a problem. We are abstracting all calls to the item table through the specific functions in the Item class. This is done since the item data is held in several tables to save space. Please use ```Item::selectThreadForUser``` for the list of threads and separate calls of ```DBA::count``` for the count. (I just realized that by now we don't have a ```Item::countForUser``` or ```Item::count```).

Reply:
**api_format_items:** 

It handle almost all APIs call, 
(ex. api_favorites_create_destroy or notification or notification or status/show/....), is that a good idea?

for example, status/show?include_entities=true, it's already can fetch comments, no necessary for this extra filed (friendica_comments)

**Item::selectThreadForUser**

It's too complicate for me, for now, if have to do so, can you help me on this? i real need this number of comments for my user.
If put it in abstracting, will cause performance issue (SQL Explain) ?

Reply:
Concerning the SQL performance: My experience is that there is not much difference between a single complicate SQL call and several - but simple - SQL calls.